### PR TITLE
Translate 404/not_found of get_file_info into FileNotPresent exception

### PR DESCRIPTION
**Pull request by elemental-lf**
Currently this return code gets translated into an UnknownError exception which is not very helpful

### CODE REVIEW

File: b2/exception.py [hunk 1]
```diff
@@ -348,9 +348,11 @@ def interpret_b2_error(status, code, message, post_params=None):
         return FileAlreadyHidden(post_params['fileName'])
     elif status == 400 and code == 'bad_json':
         return BadJson(message)
-    elif status == 400 and code in ("no_such_file", "file_not_present"):
-        # hide_file returns "no_such_file"
-        # delete_file_version returns "file_not_present"
+    elif ((status == 400 and code in ("no_such_file", "file_not_present"))
+        or (status == 404 and code in ("not_found"))):
+        # hide_file returns 400 and "no_such_file"
```
Comment (bwbeach) [inline b2/exception.py]: Thanks for the comments.  Very helpful.

**Comment by elemental-lf** [comment]
@bwbeach Thank you for feedback. I'll correct this. But there are more problems unfortunately: Further testing showed that download_file_by_id and download_file_by_id can also return 404/not_found but they don't set post_params['fileName'] because the information is part of the request URL. I'll try to fix this also.

**Comment by elemental-lf** [comment]
Regarding the AppVeyor test: The Python 3 job succeeds, but the Python 2 job fails with an error which I think isn't related to this PR:
```
Executing: python setup.py build
Traceback (most recent call last):
  File "setup.py", line 20, in <module>
    from setuptools import setup, find_packages, __version__ as setuptoolsversion
ImportError: No module named setuptools
Command exited with code 1``
```

**Comment by bwbeach** [comment]
The AppVeyor build has been fragile.

I'll take a look and see what I can figure out.

**Comment by bwbeach** [comment]
I just merged a PR to fix the appveyor build into `master`.  If you merge that into your branch, it should fix the build.

**Comment by elemental-lf** [comment]
@bwbeach, hooray, this worked out okay. I'd really like this or a similar patch to be merged, because currently I have this in my code, which is ugly:
```python
isinstance(e, FileNotPresent) or isinstance(e, UnknownError) and "404 not_found" in str(e)
```
Thank you!

**Comment by elemental-lf** [comment]
@bwbeach could this be merged please?

**Comment by bwbeach** [comment]
Sorry for the delay merging.

**Pull request merged**